rogue-thi-app/lib/backend-utils/rooms-utils.js

```diff
@@ -244,6 +249,35 @@ export async function getRoomAvailability (roomRequestList, day = new Date()) {
   return processedOpenings
 }
 
+export async function getRoomCapacity (roomRequestList, day = new Date()) {
+  const roomCapacityData = {}
+  day.setHours(0, 0, 0, 0)
+
+  const data = await API.getFreeRooms(day)
+  const thisDate = `${day.getFullYear()}-${String(day.getMonth() + 1).padStart(2, '0')}-${String(day.getDate()).padStart(2, '0')}T00:00:00.000000`
+  let dataToday
+  data.forEach(element => {
+    if (element['datum'] === thisDate) {
+      dataToday = element
+    }
+  })
+
+  if (dataToday['rtypes'] === undefined) {
+    return roomCapacityData
+  }
+  const capacityData = dataToday['rtypes']
+
+  capacityData.forEach(roomType => {
+    for (const stunde in roomType['stunden']) { //! vllt nur die erste Stunde?
+      roomType['stunden'][stunde]['raeume'].forEach(raum => {
+        roomCapacityData[raum[2]] = raum[3]
+      })
+    }
+  })
+
+  return roomCapacityData
```

Comment:
Since you have already prepared the data in `getRoomCapacity`, you can use it again here. This saves a lot of work fighting through the weird THI API format ;)

```js 
export async function getRoomCapacity (day = new Date()) {
  const data = await API.getFreeRooms(day)
  const openings = await getRoomOpenings(data, day)

  const roomCapacityData = Object.fromEntries(Object.entries(openings).map(([key, value]) => [key, value[0].capacity]))

  return roomCapacityData
}
```

The problem why a capacity is not provided for each room is that it depends on whether a room is free on that day. One way to improve this would be to search across all 7 available days